lib/helper/Puppeteer.js

```diff
@@ -513,6 +530,11 @@ class Puppeteer extends Helper {
     if (!(/^\w+\:\/\//.test(url))) {
       url = this.options.url + url;
     }
+
+    if (this.config.basicAuth) {
+      this.page.authenticate(this.config.basicAuth);
```

Comment:
1. what if we are not on the base page? If I am running `I.amOnPage('https://github.com')`
2. what if we already authenticated?

Reply:
1. make a check to see if the current is the same with base page.
2. introduce `isAuthenticated` var for checking if we are already authenticated.